### PR TITLE
Added tests for invalid BibTex Files

### PR DESCRIPTION
**Pull request by garymejia**
Tests were added as requested from the conversation in the merged pull request #10711. The tests added aim to correctly assert when the imported file returns warnings for improper citation keys and when the imported file throws an error for improper syntax in a imported bib entry. 

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/test/java/org/jabref/logic/importer/ParserResultTest.java [hunk 1]
```diff
@@ -48,4 +53,109 @@ public void doesNotHaveDuplicateWarnings() {
...
+                }
+                """;
+
+        File tempFile = File.createTempFile("invalidBibTex", ".bib");
+        try {
+            Files.write(tempFile.toPath(), bibtexEntry.getBytes());
+            ParserResult parserResult = OpenDatabase.loadDatabase(tempFile.toPath(), mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor());
+            assertEquals(parserResult.warnings().getFirst(),"Line 1: Found corrupted citation key  (contains whitespaces).");
+        } finally {
+            if (!tempFile.delete()) {
```
Comment (koppor) [inline src/test/java/org/jabref/logic/importer/ParserResultTest.java:77]: No need to delete temp files. This is handled by JUnit. Please use `@TempDir` annotation to make use of that functionality.

**Review by koppor — CHANGES_REQUESTED** [review]
Thank you for working on this. Please fix checkstyle - and use JUnit's `@TempDir`. See other code in JabRef how to use.

Could you try to remove your updates of `abbrv.jabref.org` and `csl-styles`? Maybe by merging upstream/main? Maybe, it is also possible via GitHub's UI...

**Comment by garymejia** [comment]
Sorry I just realized I didn't remove my updates to abbrv.jabref.org and csl-styles. I'll try to resolve that

**Comment by calixtus** [comment]
> Could you try to remove your updates of `abbrv.jabref.org` and `csl-styles`? Maybe by merging upstream/main? Maybe, it is also possible via GitHub's UI...

Had the same problem recently, did not work at all. Git wouldn't let me remove these changes, github ui is useless for this. Git submodules is a horrible thing, makes some things way easier and some things - argh!
I believe these submodule changes are ignored anyway when merging into main.

**Comment by garymejia** [comment]
Sorry I'm not sure what to run to discard those modified files. I tried merging to upstream but it didn't seem to help.

**Comment by calixtus** [comment]
git merge upstream/master

**Comment by calixtus** [comment]
Hm, also seems that you did not create a separate branch for your changes, but did your changes in your main branch. But this should not have an effect to this. Please remember to always create a new branch you commit your changes to, otherwise you are blocked from making any other change without further complications.

**Comment by garymejia** [comment]
Unfortunately merging upstream main doesnt remove the changed files from what I can see.

**Comment by Siedlerchr** [comment]
I fixed the issues with the submodules in https://github.com/JabRef/jabref/pull/10852

File: src/test/java/org/jabref/logic/importer/ParserResultTest.java [hunk 2]
```diff
@@ -48,4 +55,80 @@ public void doesNotHaveDuplicateWarnings() {
...
+                @article{ myArticle ,
+                  author    = "Author Name",
+                  title     = "Title of the Article",
+                  journal   = "Journal Name",
+                  year      = "2024",
+                  pages     = "1-10",
+                  publisher = "Publisher Name"
+                }
+                """;
+        File tempFile = new File(tmpDir.toFile(), "invalidBibTex.bib");
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/importer/ParserResultTest.java:72]: I fixed this in the follow up PR but you can just do:
        Path tempFile = tmpDir.resolve("invalidBibTex.bib");

**Pull request closed without merging**